app/controllers/my_module_tags_controller.rb

```diff
@@ -112,18 +114,20 @@ def destroy
 
   def search_tags
     assigned_tags = @my_module.my_module_tags.pluck(:tag_id)
-    tags = @my_module.experiment.project.tags\
-                     .where.not(id: assigned_tags)
-                     .search(
-                       current_user,
-                       false,
-                       params[:query]
-                     ).select(:id, :name, :color).limit(6)
-    tags = tags.map { |i| { value: i.id, label: i.name, params: { color: i.color } } }
-
-    if !(tags.map { |i| i[:label] }.include? params[:query]) && !params[:query].empty?
-      tags = [{ value: 0, label: params[:query], params: { color: nil } }] + tags
+    all_tags = @my_module.experiment.project.tags
+    tags = all_tags.where.not(id: assigned_tags)
+                   .search(current_user, false, params[:query])
+                   .select(:id, :name, :color)
+                   .limit(6)
+
+    tags = tags.map do |tag|
+      { value: tag.id, label: sanitize_input(tag.name), params: { color: sanitize_input(tag.color) } }
```

Comment:
Colors predefined list in constants

Reply:
And? Please give me a link to the lines where this is enforced?

Reply:
Well, yes someone can force set custom color. But what he can do with 7 characters?

Reply:
And if later this limit is changed, what will happen to this code?

Reply:
In this case yes, it can become issue